### PR TITLE
Update firmware section for Golioth Zephyr SDK v0.4.0

### PR DESCRIPTION
**Pull request by szczys**
This PR updates the firmware section to use the new API implemented in v0.4.0 of the Golioth Zephyr SDK.

### CODE REVIEW

File: docs/firmware/zephyr-device-sdk/light-db/guide-lightdb-get.md [hunk 1]
```diff
@@ -2,95 +2,92 @@
...
-static void golioth_on_message(struct golioth_client *client,
-                               struct coap_packet *rx)
-{
-    coap_response_received(rx, NULL, coap_replies,
-                           ARRAY_SIZE(coap_replies));
-}
-```
+The returned value (or JSON object) is stored in a character array. A pointer
+to this array and the length of the array are passed as the final two
+parameters of the synchronous get function.
```
Comment (ncmiller) [inline docs/firmware/zephyr-device-sdk/light-db/guide-lightdb-get.md:41]: It's probably worth pointing out that the `len` parameter is both an input and an output.

From Doxygen:

```
 * @param[in,out] len Size of buffer on input, size of response on output
```

File: docs/firmware/zephyr-device-sdk/light-db/guide-lightdb-get.md [hunk 2]
```diff
@@ -2,95 +2,92 @@
...
 
-The `golioth_lightdb_get()` API call requests data from the `counter` endpoint,
-passing a `coap_reply` object and a callback function to execute when the reply
-is received from Golioth. The `coap_reply_next_unused()` function ensures that
-the `coap_reply` struct being passed is not already in use by another request.
+The response is passed to the callback function as a `golioth_req_rsp` struct
+that includes `err`, `data`, and `len`. It is recommended that callbacks test
+for an error code to ensure the expected data was received.
+
+User-defined arguments are passed to the callback as `rsp.user_data`.
```
Comment (ncmiller) [inline docs/firmware/zephyr-device-sdk/light-db/guide-lightdb-get.md:85]: `rsp->user_data`

File: docs/firmware/zephyr-device-sdk/light-db/guide-lightdb-set.md [hunk 3]
```diff
@@ -5,39 +5,101 @@ sidebar_position: 4
 
 The [LightDB Set sample
 application](https://github.com/golioth/golioth-zephyr-sdk/tree/main/samples/lightdb/set)
-demonstrates how to send resources to LightDB State using the Golioth SDK. The
-process in non-blocking and after submitting a set request, the Golioth system
-client will complete the operation asyncronously.
+demonstrates how to send resources to LightDB State using the Golioth SDK. Data
+can be set synchronously, or asyncronously. Synchronous requests block program
```
Comment (ncmiller) [inline docs/firmware/zephyr-device-sdk/light-db/guide-lightdb-set.md:9]: `asynchronously`

File: docs/firmware/zephyr-device-sdk/light-db/guide-lightdb-set.md [hunk 4]
```diff
@@ -5,39 +5,101 @@ sidebar_position: 4
...
+```
+
+Including the Golioth System Client header file makes the LightDB Set API
+functions available to your program.
+
+## Payload preparation
 
 ```c
-char sbuf[64]; // Large enough for data plus zero terminator
+char sbuf[11]; // Large enough for data plus zero terminator
```
Comment (ncmiller) [inline docs/firmware/zephyr-device-sdk/light-db/guide-lightdb-set.md:25]: Nit: prefer using the term "NULL terminator". They mean the same thing, but I think NULL terminator is more common.

**Review by ncmiller — COMMENTED** [review]
Looks great so far, just some minor suggestions

**Review by ncmiller — APPROVED** [review]
LGTM!

**Review by mniestroj — APPROVED** [review]

**Pull request merged**